ServerCore/Pages/Teams/Hints.cshtml

```diff
@@ -90,16 +91,25 @@ You have <b>
         </tr>
     </thead>
     <tbody>
-        @foreach (var hint in Model.Hints)
+        @foreach (var hint in Model.HintViewStates)
         {
+            var hintViewState = hint as HintsModel.HintViewState;
+            var puzzleThreadViewState = hint as HintsModel.PuzzleThreadViewState;
             <tr>
                 <td>
-                    @hint.Hint.Description
+                    @hint.Description
                 </td>
                 <td>
                     @if (hint.IsUnlocked)
                     {
-                        @hint.Hint.Content
+                        if (hintViewState != null)
+                        {
+                            @hint.Content
+                        }
+                        else if (puzzleThreadViewState != null)
+                        {
+                            <a asp-Page="/Threads/PuzzleThread" asp-route-puzzleid="@Model.PuzzleID" asp-route-teamId="@Model.Team?.ID" asp-route-playerId="@Model.LoggedInUser.ID">Use this link to message author!</a>
```

Comment:
If there's room, "Use this link to message **the** author!" would be clearer.

Reply:
done!